Origen: https://www.kaggle.com/datasets/aungpyaeap/supermarket-sales/data

In [2]:
import pandas as pd
import numpy as np
import matplotlib as plt

In [3]:
df = pd.read_csv("supermarket_sales - Sheet1.csv")

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 17 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   Invoice ID               1000 non-null   object 
 1   Branch                   1000 non-null   object 
 2   City                     1000 non-null   object 
 3   Customer type            1000 non-null   object 
 4   Gender                   1000 non-null   object 
 5   Product line             1000 non-null   object 
 6   Unit price               1000 non-null   float64
 7   Quantity                 1000 non-null   int64  
 8   Tax 5%                   1000 non-null   float64
 9   Total                    1000 non-null   float64
 10  Date                     1000 non-null   object 
 11  Time                     1000 non-null   object 
 12  Payment                  1000 non-null   object 
 13  cogs                     1000 non-null   float64
 14  gross margin percentage  

In [5]:
df.describe()

Unit price     Quantity       Tax 5%        Total        cogs  \
count  1000.000000  1000.000000  1000.000000  1000.000000  1000.00000   
mean     55.672130     5.510000    15.379369   322.966749   307.58738   
std      26.494628     2.923431    11.708825   245.885335   234.17651   
min      10.080000     1.000000     0.508500    10.678500    10.17000   
25%      32.875000     3.000000     5.924875   124.422375   118.49750   
50%      55.230000     5.000000    12.088000   253.848000   241.76000   
75%      77.935000     8.000000    22.445250   471.350250   448.90500   
max      99.960000    10.000000    49.650000  1042.650000   993.00000   

       gross margin percentage  gross income      Rating  
count             1.000000e+03   1000.000000  1000.00000  
mean              4.761905e+00     15.379369     6.97270  
std               6.131498e-14     11.708825     1.71858  
min               4.761905e+00      0.508500     4.00000  
25%               4.761905e+00      5.924875     5.50000  
50%               4.761905e+00     12.088000     7.00000  
75%               4.761905e+00     22.445250     8.50000  
max               4.761905e+00     49.650000    10.00000

In [6]:
df.head(5)

Invoice ID Branch       City Customer type  Gender  \
0  750-67-8428      A     Yangon        Member  Female   
1  226-31-3081      C  Naypyitaw        Normal  Female   
2  631-41-3108      A     Yangon        Normal    Male   
3  123-19-1176      A     Yangon        Member    Male   
4  373-73-7910      A     Yangon        Normal    Male   

             Product line  Unit price  Quantity   Tax 5%     Total       Date  \
0       Health and beauty       74.69         7  26.1415  548.9715   1/5/2019   
1  Electronic accessories       15.28         5   3.8200   80.2200   3/8/2019   
2      Home and lifestyle       46.33         7  16.2155  340.5255   3/3/2019   
3       Health and beauty       58.22         8  23.2880  489.0480  1/27/2019   
4       Sports and travel       86.31         7  30.2085  634.3785   2/8/2019   

    Time      Payment    cogs  gross margin percentage  gross income  Rating  
0  13:08      Ewallet  522.83                 4.761905       26.1415     9.1  
1  10:29         Cash   76.40                 4.761905        3.8200     9.6  
2  13:23  Credit card  324.31                 4.761905       16.2155     7.4  
3  20:33      Ewallet  465.76                 4.761905       23.2880     8.4  
4  10:37      Ewallet  604.17                 4.761905       30.2085     5.3

Preguntas como punto de partida para análisis:

¿Cuál supermercado vendió más?
¿Cuál producto fue el más vendido y cuál fue el menos vendido?
¿Qué género compró más y de qué producto en particular?
¿Qué forma de pago fue la más usada?

ML:
Que ofertas generar?
a que público apuntarlas?
que medios de pago incentibar? 

In [16]:
#Cuál supermercado vendió mas?
ventas_por_supermercado = df.groupby('Branch')['Quantity'].sum()

# Resultado
print(ventas_por_supermercado)
print("El Supermercado que más vendió fue: el A ", ventas_por_supermercado.max())

Branch
A    1859
B    1820
C    1831
Name: Quantity, dtype: int64
El Supermercado que más vendió fue: el A  1859


In [18]:
#Cuál producto fue el mas vendido y cual el menos vendido?
producto_mas_vendido = df.groupby("Product line")['Quantity'].sum()

In [19]:
print("Los Productos más y menos vendidos fueron: ")
print("Electronic accessories: ", producto_mas_vendido.max())
print("Health and beauty: ", producto_mas_vendido.min())

Los Productos más y menos vendidos fueron: 
Electronic accessories:  971
Health and beauty:  854


In [24]:
#Que género compró más y de que producto?

productos_mas_comprados_por_genero = df.groupby(['Gender', 'Product line']).size().reset_index(name='Cantidad')

productos_mas_comprados_por_genero_max = productos_mas_comprados_por_genero.loc[productos_mas_comprados_por_genero.groupby('Gender')['Cantidad'].idxmax()]
print(productos_mas_comprados_por_genero_max)

   Gender         Product line  Cantidad
1  Female  Fashion accessories        96
9    Male    Health and beauty        88


In [15]:
#Cuál fue la forma de pago más utilizada?
forma_mas_utilizada = df.groupby("Payment").size().sort_values(ascending=False)
forma_mas_utilizada

Payment
Ewallet        345
Cash           344
Credit card    311
dtype: int64

# Conclusión

Tras analizar detalladamente las ventas de las tres tiendas de nuestro supermercado, se ha observado que no existen diferencias significativas en cuanto a las ventas entre ellas. Sin embargo, al examinar los productos más y menos vendidos, se destacan los artículos de Electronic accessories como los más demandados, con un total de 971 unidades vendidas, mientras que los productos de Health and beauty registraron la menor cantidad de ventas, con 854 unidades. Al profundizar en el análisis por género, se identificó que las mujeres mostraron una preferencia por los productos de Fashion accessories, adquiriendo un total de 96 unidades, mientras que los hombres demostraron una mayor inclinación hacia los productos de Health and beauty, con un total de 88 unidades vendidas.

En cuanto a las formas de pago, se observa que Ewallet y Cash fueron las opciones más utilizadas, con 345 y 344 transacciones respectivamente, seguidas de cerca por las transacciones con tarjeta de crédito, que totalizaron 311.

En resumen, para maximizar las ganancias y satisfacer las preferencias de nuestros clientes, recomendamos ofrecer promociones especiales orientadas a las preferencias por género, así como descuentos en los productos menos vendidos, junto con facilidades en relación a las formas de pago más utilizadas. Esta estrategia podría contribuir significativamente a aumentar las ventas y mejorar la experiencia de compra de nuestros clientes en todas nuestras tiendas.